File: output/gemma4-e4b/avatar/Java/Python/codeforces_108_B.py
import sys

def solve():
    # Read all input tokens from standard input
    data = sys.stdin.read().split()
    
    if not data:
        return

    try:
        # The first element is N
        n = int(data[0])
    except ValueError:
        return

    # The remaining elements are the array A
    if len(data) < n + 1:
        return

    a = [int(x) for x in data[1:n+1]]

    # Sort the array
    a.sort()

    # Check the condition
    found = False
    for i in range(len(a) - 1):
        # Check if a[i+1] < a[i] * 2 AND a[i] != a[i+1]
        if a[i+1] < a[i] * 2 and a[i] != a[i+1]:
            print("YES")
            found = True
            break
    
    if not found:
        print("NO")

if __name__ == "__main__":
    solve()

File: output/gemma4-e4b/avatar/Python/Java/codeforces_108_B.java
import java.util.Arrays;
import java.util.Scanner;

public class codeforces_108_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        if (!scanner.hasNextLine()) {
            System.out.println("NO");
            return;
        }
        
        String line = scanner.nextLine();
        String[] parts = line.trim().split("\\s+");
        
        if (parts.length == 0 || (parts.length == 1 && parts[0].isEmpty())) {
            System.out.println("NO");
            return;
        }

        int[] a = new int[parts.length];
        for (int i = 0; i < parts.length; i++) {
            a[i] = Integer.parseInt(parts[i]);
        }
        
        Arrays.sort(a);
        
        boolean found = false;
        for (int i = 1; i < a.length; i++) {
            // Check the condition: a[i] != a[i - 1] AND a[i] < a[i - 1] * 2
            if (a[i] != a[i - 1] && a[i] < a[i - 1] * 2) {
                found = true;
                break;
            }
        }
        
        if (found) {
            System.out.println("YES");
        } else {
            System.out.println("NO");
        }
    }
}